addons/purchase/purchase.py

```diff
@@ -1423,21 +1403,30 @@ def _get_po_line_values_from_procs(self, cr, uid, procurements, partner, schedul
                 fp = acc_pos_obj.browse(cr, uid, fp, context=context)
             taxes = acc_pos_obj.map_tax(cr, uid, fp, taxes_ids)
             name = names_dict[procurement.product_id.id]
+            uom_id = procurement.product_id.uom_po_id.id
+            qty = uom_obj._compute_qty(cr, uid, procurement.product_uom.id, procurement.product_qty, uom_id)
+            price = procurement.product_id.standard_price
             if procurement.product_id.description_purchase:
                 name += '\n' + procurement.product_id.description_purchase
-            price = prices[procurement.product_id.id][pricelist_id]
-            price = uom_obj._compute_price(cr, uid, procurement.product_uom.id, price, to_uom_id=procurement.product_id.product_tmpl_id.uom_po_id.id)
-
             values = {
                 'name': name,
-                'product_qty': qty,
                 'product_id': procurement.product_id.id,
                 'product_uom': procurement.product_id.uom_po_id.id,
-                'price_unit': price or 0.0,
                 'date_planned': schedule_date.strftime(DEFAULT_SERVER_DATETIME_FORMAT),
                 'taxes_id': [(6, 0, taxes)],
                 'procurement_ids': [(4, procurement.id)]
                 }
+            # calculate price/qty for valid vendor
+            for supplier in procurement.product_id.seller_ids:
```

Comment:
do not duplicate logic to find best supplier . I think, need make one function for that.